packages/block-editor/src/components/block-list/insertion-point.js

```diff
@@ -107,56 +46,123 @@ function InsertionPointPopover( {
 	containerRef,
 	showInsertionPoint,
 } ) {
-	const element = useSelect(
+	const { previousElement, nextElement, orientation, isHidden } = useSelect(
 		( select ) => {
-			const { getBlockOrder } = select( 'core/block-editor' );
+			const {
+				getBlockOrder,
+				getBlockRootClientId,
+				getBlockListSettings,
+				getMultiSelectedBlockClientIds,
+				getSelectedBlockClientId,
+				hasMultiSelection,
+				getSettings,
+			} = select( 'core/block-editor' );
 			const { ownerDocument } = containerRef.current;
-			const targetClientId =
-				clientId || last( getBlockOrder( rootClientId ) );
+			const targetRootClientId = clientId
+				? getBlockRootClientId( clientId )
+				: rootClientId;
```

Comment:
I was wondering why the logic wasn't something like:
```jsx
const targetRootClientId = rootClientId ? rootClientId : getBlockRootClientId( clientId );
```

I'm sure there's a good reason, but might be worth leaving a comment for others reading the code.

Reply:
It has to do with the fact that this is both used for the insertion point when the inserter is opened (use the insertion poin t from there) and the hovered insertion point.